### PR TITLE
extending creative commons model

### PR DESCRIPTION
**Pull request by jamesgorrie**
The current model of creative commons is not extensive enough - this adds:
* `licence`
* `source`
* `creator`
* `contentLink`

And Maps to:
* `credit => licence / source`
* `creator => byline`
* `contentLink => creditLink`

- [ ] Get actual list from RH for licenses (I've included the current commercial licences)
- [ ] Do mappings to metadata

### CODE REVIEW

File: common-lib/src/main/scala/com/gu/mediaservice/model/UsageRights.scala [hunk 1]
```diff
@@ -396,5 +397,12 @@ case class CreativeCommons(restrictions: Option[String] = None)
   }
 object CreativeCommons {
  implicit val jsonReads: Reads[CreativeCommons] = Json.reads[CreativeCommons]
- implicit val jsonWrites: Writes[CreativeCommons] = UsageRights.defaultWrites
+ implicit val jsonWrites: Writes[CreativeCommons] = (
```
Comment (theefer) [inline common-lib/src/main/scala/com/gu/mediaservice/model/UsageRights.scala]: Why does it need to be explicit now?

Comment (jamesgorrie) [inline common-lib/src/main/scala/com/gu/mediaservice/model/UsageRights.scala]: As the `UsageRights.defaultRights` can only deals with category and restrictions.
I have a branch looking at composing said function (or using partials) which cleans up a lot of this manual labour - I just haven't had time to completely finish / figure it out.

Comment (jamesgorrie) [inline common-lib/src/main/scala/com/gu/mediaservice/model/UsageRights.scala]: We can't use Json.writes[CreativeCommons] either as that then leaves out category et al as it only takes in the parameters of the case class.

**Comment by theefer** [comment]
:+1:

**Pull request merged**